### PR TITLE
Us 17 18

### PR DESCRIPTION
**Pull request by helloeduardo**
- [x] done

User Story 17, Items Index Page

As any kind of user on the system
I can visit the items catalog ("/items")
I see all items in the system except disabled items

The item image is a link to that item's show page

- [x] done

User Story 18, Items Index Page Statistics

As any kind of user on the system
When I visit the items index page ("/items")
I see an area with statistics:
- the top 5 most popular items by quantity purchased, plus the quantity bought
- the bottom 5 least popular items, plus the quantity bought

"Popularity" is determined by total quantity of that item ordered

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -12,6 +12,21 @@ class Item <ApplicationRecord
...
   validates_numericality_of :price, greater_than: 0
 
+  def self.top_five
+      joins(:item_orders)
+      .select('items.id', 'sum(item_orders.quantity) AS total', 'items.name')
+      .group('items.id')
+      .order('total desc')
+      .limit(5)
+  end
+
```
Comment (BJSherman80) [inline app/models/item.rb:22]: nice AR query methods with some injected sql

File: spec/features/items/index_spec.rb [hunk 2]
```diff
@@ -47,15 +46,63 @@
...
+      expect(page).to_not have_content("Inventory: #{@dog_bone.inventory}")
+      expect(page).to_not have_css("img[src*='#{@dog_bone.image}']")
+    end
+
+    it "I can click image to go to item show page" do
+      visit '/items'
+
+      within "#item-#{@tire.id}" do
+        find(".image-link").click
+      end
```
Comment (BJSherman80) [inline spec/features/items/index_spec.rb:62]: Interesting syntax here for clicking an image for a link

File: spec/features/items/index_spec.rb [hunk 3]
```diff
@@ -47,15 +46,63 @@
...
+          expect(page.all('li')[4].text).to eq("Pull Toy: Quantity Bought: 5")
+        end
+
+        within ".bottom-five" do
+          expect(page.all('li')[0].text).to eq("Gatorskins: Quantity Bought: 2")
+          expect(page.all('li')[1].text).to eq("Pull Toy: Quantity Bought: 5")
+          expect(page.all('li')[2].text).to eq("Chain: Quantity Bought: 6")
+          expect(page.all('li')[3].text).to eq("Dog Bone: Quantity Bought: 7")
+          expect(page.all('li')[4].text).to eq("Lizard Head: Quantity Bought: 8")
+        end
```
Comment (BJSherman80) [inline spec/features/items/index_spec.rb:102]: Good job at testing for the order here

**Review by BJSherman80 — APPROVED** [review]

**Review by nmking22 — APPROVED** [review]
Looks great! Nice clean code for the top_five and bottom_five methods

**Review by the-color-bliu — APPROVED** [review]
Nice before :each in the testing, everything is looking here, merge approved!

**Pull request merged**